src/common/proc.c

```diff
@@ -163,7 +163,16 @@ crt_proc_daos_prop_t(crt_proc_t proc, daos_prop_t **data)
 		*data = prop;
 		return rc;
 	case CRT_PROC_FREE:
-		daos_prop_free(*data);
+		prop = *data;
+		if (prop == NULL)
+			return 0;
+		if (prop->dpp_nr == 0 || prop->dpp_entries == NULL) {
+			D_FREE_PTR(prop);
```

Comment:
We should have switched to D_FREE() and be removing D_FREE_PTR() but that can wait for a cleanup patch.